doc/design.md

```diff
@@ -442,3 +500,356 @@ Covers:
 * [`req~npm-changed-dependency~1`](system_requirements.md#get-changed-dependency)
 
 Needs: impl, utest, itest
+
+## Automatic Dependency Update Process
+
+This consists of the following steps:
+1. Trigger the dependency update process
+2. Update dependencies
+3. Create a pull request
+
+![Activity Diagram for the dependencies update process](images/dependencies_update_process.svg)
+
+#### Triggering the Dependency Update Process
+`dsn~trigger-dependency-updates~1`
+
+PK generates the `dependencies_check.yml` GitHub workflow so that it launches the `dependencies_update.yml` workflow when it detects new vulnerabilities.
+
+Rationale:
+
+`dependencies_check.yml` already uses the [security-issues](https://exasol.github.io/python-toolbox/github_actions/security_issues.html) tool from the [python-toolbox](https://github.com/exasol/python-toolbox) to create issues for new vulnerabilities. Re-implementing this in PK is not necessary.
+
+Covers:
+* [`req~auto-update-dependencies~1`](system_requirements.md#auto-update-dependencies)
+
+-Needs: impl, utest, itest
+
+#### Update Dependencies Mode
+`dsn~update-dependencies-mode~1`
+
+PK provides an `update-dependencies` mode in addition to `fix` and `verify`. This mode performs the following steps:
+
+1. Increment version of the project
+2. Update dependencies to their latest version
+3. Create changelog containing information about the fixed vulnerabilities (if available)
+
+Rationale:
+
+* We implement this in PK because
+  * PK already contains code for working with versions and changelog, so we can reuse this code
+  * The `update-dependencies` mode is also useful for running locally on the developer's machine when working on a non-security related task
+* We don't implement git/GitHub operations in PK because
+  * This would couple PK to GitHub
+  * This would be surprising when running it locally
+  * This would require credentials for accessing the GitHub API
+
+Covers:
+* [`req~auto-update-dependencies~1`](system_requirements.md#auto-update-dependencies)
+* [`req~auto-create-changelog~1`](system_requirements.md#automatically-create-change-log-entry)
+
+Needs: dsn
+
+##### Incrementing the Project Version
+`dsn~increment-version~1`
+
+PK increments the project's patch version. PK does not modify the version if the current version was not yet released (i.e. there is not release in the latest changelog file).
```

Comment:
That means, the human needs to increase this version afterward manually when they want to release. I think this fair. Only question, assuming the current unreleased changelog file points to version 1.2.1 , the previous release version was 1.2.0 and now a dependency update comes in, which version will PK choose, 1.2.2 or will it replace 1.2.1?

Reply:
PK will check the release date of the 1.2.1 changes file. If it is undefined or today, PK assumes 1.2.1 was not released and won't increment the version. PK will update the changes file for 1.2.1 with the security update.